### PR TITLE
GAWB-3830: Pass workflow collections and submission and workspace ids to CromIAM

### PR DESCRIPTION
**Pull request by ansingh7115**
Companion: https://github.com/broadinstitute/firecloud-develop/pull/1452

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala [hunk 1]
```diff
@@ -290,6 +294,13 @@ trait WorkflowSubmission extends FutureSupport with LazyLogging with MethodWiths
         wdl <- getWdl(methodConfig, userCredentials)
       } yield {
         val wfOpts = buildWorkflowOpts(workspaceRec, submissionRec.id, RawlsUser(RawlsUserSubjectId(userIdInfo), RawlsUserEmail(submissionRec.submitterEmail)), petSAJson, billingProject, submissionRec.useCallCache, WorkflowFailureModes.withNameOpt(submissionRec.workflowFailureMode))
+        val submissionAndWorkspaceLabels = Map("submission-id" -> submissionRec.id.toString,  "workspace-id" -> workspaceRec.id.toString)
+        val wfLabels = workspaceRec.workflowCollection match {
+          case None => submissionAndWorkspaceLabels
+          case Some(workflowCollection) => if (useWorkflowCollectionField) submissionAndWorkspaceLabels else submissionAndWorkspaceLabels + ("caas-collection-name" -> workflowCollection)
+        }
+        val wfCollection = if (useWorkflowCollectionField) workspaceRec.workflowCollection else None
+
```
Comment (ansingh7115) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala:306]: I chose `submission-id` and `workspace-id` for the labels and the [ticket ](https://broadinstitute.atlassian.net/browse/GAWB-3830)seemed to specify `caas-collection-name` for the workflow collection label name. I'm unsure why is specifies "caas"?

Comment (helgridly) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala:306]: cromwell as a service. the cromfolks made that up

File: core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala [hunk 2]
```diff
@@ -185,12 +185,13 @@ object Boot extends App with LazyLogging {
 
     val maxActiveWorkflowsTotal = conf.getInt("executionservice.maxActiveWorkflowsPerServer") * executionServiceServers.size
     val maxActiveWorkflowsPerUser = maxActiveWorkflowsTotal / conf.getInt("executionservice.activeWorkflowHogFactor")
+    val useWorkflowCollectionField = conf.getBoolean("executionservice.useWorkflowCollectionField")
```
Comment (helgridly) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala:188]: probably good to doublecheck whether we might want to have _both_ at some point during the migration, in which case this won't work

**Review by helgridly — APPROVED** [review]
one thing to doublecheck but LGTM

File: core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala (hunk 1 above)
Comment (ansingh7115) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala:306]: So can I just call it "workflow-collection"?

File: core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala (hunk 2 above)
Comment (ansingh7115) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala:188]: I could add 2 separate flags instead

File: core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala (hunk 1 above)
Comment (helgridly) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala:306]: the label is read and interpreted by cromiam, so you have to call it what they're looking for

**Review by dvoet — APPROVED** [review]

**Comment by helgridly** [comment]
closed for #1027

**Pull request closed without merging**